src/app/shared/document/document-detail/document-detail.component.ts

```diff
@@ -0,0 +1,401 @@
+
+    this.createdBy$ = this.store.select(selectUserNameById(this._documentModel.createdBy))
+      .pipe(filter(name => !isNullOrUndefined(name)));
+    this.updatedBy$ = this.store.select(selectUserNameById(this._documentModel.updatedBy))
+      .pipe(filter(name => !isNullOrUndefined(name)));
+
+    this.subscriptions.add(this.store.select(selectOrganizationByWorkspace)
+      .pipe(filter(org => !isNullOrUndefined(org)), take(1))
+      .subscribe(org => this.store.dispatch(new UsersAction.Get({ organizationId: org.id} ))));
+
+    this.encodeDocument();
+  }
+
+  private encodeDocument() {
+    this.summary = this.getDocumentSummary();
+
+    this.collection.attributes.forEach(attr => {
+      if (attr.usageCount > 0 && this._documentModel.data[attr.id] !== undefined) {
+        let row = this.getRowById(attr.id);
+        if (row) {
+          row.remove = false;
+          if (row.name !== attr.name) {
+            row.name = attr.name;
+          }
+          if (row.value !== this._documentModel.data[attr.id]) {
+            row.value = this._documentModel.data[attr.id];
+          }
+        } else {
+          row = this.getRowByCorrelationId(attr.correlationId);
+          if (row) {
+            row.remove = false;
+            row.id = attr.id;
+            if (row.name !== attr.name) {
+              row.name = attr.name;
+            }
+            if (row.value !== this._documentModel.data[attr.id]) {
+              row.value = this._documentModel.data[attr.id];
+            }
+          } else {
+            this.rows.push({ id: attr.id, name: attr.name, value: this._documentModel.data[attr.id], correlationId: attr.correlationId });
+          }
+        }
+      } else {
+        this.removeRowById(attr.id);
+      }
+    });
+  }
+
+  private getRowById(id: string): DetailRow {
+    if (id) {
+      return this.rows.find(row => row.id === id);
+    }
+
+    return null;
+  }
+
+  private getRowByCorrelationId(correlationId: string): DetailRow {
+    if (correlationId) {
+      return this.rows.find(row => row.correlationId === correlationId);
+    }
+
+    return null;
+  }
+
+  private getRowByName(name: string): DetailRow {
+    if (name) {
+      return this.rows.filter(row => !row.remove).find(row => row.name === name);
+    }
+
+    return null;
+  }
+
+  private getRemovableRowByName(name: string): DetailRow {
+    if (name) {
+      return this.rows.filter(row => row.remove).find(row => row.name === name);
+    }
+
+    return null;
+  }
+
+  private removeRowById(id: string) {
+    const idx = this.rows.indexOf(this.getRowById(id));
+    if (idx >= 0 && (this.rows[idx].remove || !this._documentModel.data[id])) {
+      this.rows.splice(idx, 1);
+    }
+  }
+
+  public getNativeDate(dateObject): number {
+    if (dateObject) {
+      return new Date(dateObject.year, dateObject.monthValue, dateObject.dayOfMonth, dateObject.hour, dateObject.minute, dateObject.second).getTime();
```

Comment:
We should format date properly on the back-end. We haven't do it yet since we did not need to show it on the front-end before.

Reply:
We use LocalDateTime on the backend, this is a huge problem at first place!!!